packages/init/src/node-async-local-storage-patch.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-underscore-dangle */
+
+import { AsyncLocalStorage, executionAsyncResource } from 'node:async_hooks';
+
+const { is: ObjectIs } = Object;
+const { apply: ReflectApply } = Reflect;
+
+/**
+ * @typedef {object} AsyncLocalStorageInternal
+ * @property {WeakMap} _resourceStoreMap
+ * @property {boolean} enabled
+ * @property {typeof _propagate} _propagate
+ * @property {(this: AsyncLocalStorage) => void} _enable
+ */
+
+Object.defineProperty(AsyncLocalStorage.prototype, 'kResourceStore', {
+  configurable: true,
+  /**
+   * @this {AsyncLocalStorage & AsyncLocalStorageInternal}
+   */
+  set() {
+    this._resourceStoreMap = new WeakMap();
+  },
+});
+
+/**
+ * @this {AsyncLocalStorage & AsyncLocalStorageInternal}
+ * @param {object} resource
+ * @param {object} triggerResource
+ * @param {string} [type]
+ */
+function _propagate(resource, triggerResource, type) {
+  if (!this.enabled) return;
+
+  const storeMap = this._resourceStoreMap;
+  storeMap.set(resource, storeMap.get(triggerResource));
+}
```

Comment:
Might as well follow our normal lexical conventions rather than exposing a new property.
```suggestion
const resourceStoreMaps = new WeakMap();

Object.defineProperty(AsyncLocalStorage.prototype, 'kResourceStore', {
  configurable: true,
  /**
   * @this {AsyncLocalStorage & AsyncLocalStorageInternal}
   */
  set() {
    resourceStoreMaps.set(this, new WeakMap());
  },
});

/**
 * @this {AsyncLocalStorage & AsyncLocalStorageInternal}
 * @param {object} resource
 * @param {object} triggerResource
 * @param {string} [type]
 */
function _propagate(resource, triggerResource, type) {
  if (!this.enabled) return;

  const storeMap = resourceStoreMaps.get(this);
  if (storeMap) storeMap.set(resource, storeMap.get(triggerResource));
}
```
etc.